### ISSUE TITLE
[Feature Request] Support for tests

### ISSUE THREAD

**Issue by SandroDamilano** [catawiki/rate-limit#15]
tl;dr: It would be great to provide a way to run tests without worrying the gem will prevent consecutive runs to be successful. This could be provided in the form of a specific method/implementation enhancement or at least with some guidance in the docs.

## Problem
I've experienced in my project that some tests when run twice in a row they fail in the second run. This happens because `RateLimit.throttle!` is failing the second time.
To solve the problem, as we're using Redis, I've added `Redis.new.flushall` to the `before` statement in our `rails_helper`.

## Solution
It would be great if there was a way to configure the gem itself to prevent this from happening, like setting it to test mode.
Otherwise, maybe at least adding some guidance to the docs on how to deal with such scenario would also be fine.

**Comment by dhruvCW** [comment]
>It would be great if there was a way to configure the gem itself to prevent this from happening, like setting it to test mode.

I don't know how this would work, especially since test scenarios can be very diverse.

IMO a good option would be to introduce 

```ruby
RateLimit.clear_all_counters
```

which for now would just wrap `RateLimit.config.flushall` but abstract away the internal mechanisms from the tests.

**Comment by mohameddiaa27** [comment]
Thanks for the request. I'm going to add some new issues to support multiple cache adaptors and improvement for the overall caching layer. Then we can pick this issue during that milestone 👍
